app/screens/channel_members/index.js

```diff
@@ -7,16 +7,23 @@ import {connect} from 'react-redux';
 import {handleRemoveChannelMembers} from 'app/actions/views/channel_members';
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 import {getCurrentChannel, canManageChannelMembers} from 'mattermost-redux/selectors/entities/channels';
-import {getProfilesInCurrentChannel} from 'mattermost-redux/selectors/entities/users';
+import {makeGetProfilesInChannel} from 'mattermost-redux/selectors/entities/users';
 import {getProfilesInChannel, searchProfiles} from 'mattermost-redux/actions/users';
 
 import ChannelMembers from './channel_members';
 
 function mapStateToProps(state) {
```

Comment:
It's still generating a new `getChannelMembers` function on each call to this so it won't be memoizing the results at all, but what you can do is change this to a `makeMapStateToProps` and make the function there so that each copy of `mapStateToProps` will have it's own function

```javascript
function makeMapStateToProps() {
    const getChannelMembers = makeGetProfilesInChannel();

    return (state) => {
        const currentChannelMembers = getChannelMembers(state, currentChannel.id, true);

        ...
    };
}
```

Reply:
same comment as @hmhealey

Reply:
Thanks guys, hope this is all good now.